### PR TITLE
Fourier drawing functionality

### PR DESCRIPTION
**Pull request by martian17**
Implemented the fourier transform function, and added some presets that makes use of that.

### CODE REVIEW

File: src/pages/FourierAnalysis/constants.js [hunk 1]
```diff
@@ -1,21 +1,98 @@
...
+    const n = 4;//number of periods
+    for(let i = 0; i < width; i++){
+      const r = i/width;
+      arr.push(Math.abs((r*n)%1*2-1)*2-1);
+    }
+    return {
+      points:arr,
+      degree:50
+    };
+    /*return {
```
Comment (zigen) [inline src/pages/FourierAnalysis/constants.js]: if you don't need this, could you remove it?

File: src/pages/FourierAnalysis/constants.js [hunk 2]
```diff
@@ -1,21 +1,98 @@
...
+    return {
+      points:[0,0,0,0,0,0],
+      degree:10
+    }
+  },
+  "AQUA": () => {
+    //temp canvas to trace the letters
+    const canvas = document.createElement("canvas");
+    //document.body.appendChild(canvas);
+    const width = 1500;
```
Comment (zigen) [inline src/pages/FourierAnalysis/constants.js]: hmm, we should need to handle the "resize" event later.

File: src/pages/FourierAnalysis/constants.js [hunk 3]
```diff
@@ -1,21 +1,98 @@
...
+  },
+  "Flat": () => {
+    return {
+      points:[0,0,0,0,0,0],
+      degree:10
+    }
+  },
+  "AQUA": () => {
+    //temp canvas to trace the letters
+    const canvas = document.createElement("canvas");
```
Comment (zigen) [inline src/pages/FourierAnalysis/constants.js]: it's obviously deviated from the React way. let's refactor this later (in another PR)

File: src/pages/FourierAnalysis/constants.js [hunk 4]
```diff
@@ -1,21 +1,98 @@
...
+    //document.body.appendChild(canvas);
```
Comment (zigen) [inline src/pages/FourierAnalysis/constants.js]: if you don't need this, don't comment out. just delete it. please...

File: src/pages/FourierAnalysis/waveForm.js [hunk 5]
```diff
@@ -1,48 +1,90 @@
...
-  export default WaveForm;
\ No newline at end of file
+  fourierTransform() {
+    //does fourier transform based on this.points
+    const harmonics = [];
+    for (let i = 0; i < this.degree; i++) {
+      harmonics.push(this.computeNthTerm(i));
+    }
+    console.log(this.points);
+    console.log(harmonics);
```
Comment (zigen) [inline src/pages/FourierAnalysis/waveForm.js]: remove all `console.log` for debug purposes. basically, we don't need it.

**Review by zigen — CHANGES_REQUESTED** [review]
it works on my machine, and it's so nice! almost LGTM.
just I left several nitpicks, could you check and consider fixing it?

File: src/pages/FourierAnalysis/constants.js (hunk 2 above)
Comment (martian17) [reply src/pages/FourierAnalysis/constants.js]: oh actually this has nothing to do with resizing, I'm creating a height map by drawing letters to a temp canvas.
The canvas will be created and destroyed in an instant, and will never be displayed.

Comment (martian17) [reply src/pages/FourierAnalysis/constants.js]: I reckon it's kind of hacky though, it's basically contained so shouldn't have any side effects.

Comment (zigen) [reply src/pages/FourierAnalysis/constants.js]: Ah, I totally understood what you want to do. why don't you separate the text image generation into another function on top of this file?

**Review by zigen — APPROVED** [review]
LGTM

**Pull request merged**